File: Is Subsequence/is_subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        s_cnt, t_cnt = 0, 0 # declare pointers
        s_len, t_len = len(s), len(t) # get lengths
        while s_cnt < s_len and t_cnt < t_len:
            if s[s_cnt] == t[t_cnt]: # check equality of two letters
                s_cnt += 1 # if two are the same, iterate both pointers
                t_cnt += 1
            else: # if they're not the same
                t_cnt += 1 # only iterate string we're trying to find subsequence in

        return s_cnt == len(s) # if s never gets to the same length, we know no subsequence

if __name__ == "__main__":
    sol = Solution()

    print(sol.isSubsequence("abc", "ahdbedc"))

File: Is Subsequence/is_subsequence.c
#include <stdio.h>
#include <string.h>

int isSubsequence(char * s, char * t){
    int s_cnt, t_cnt = 0; // grab pointers and size
    int size_s = strlen(s);
    int size_t = strlen(t);

    while(s_cnt < size_s && t_cnt < size_t){ // while the index's are valid
        if(s[s_cnt] == t[t_cnt]) // if two char's are the same
            s_cnt++; // increment s pointer
        t_cnt++; // increment t pointer
    }
    return s_cnt == size_s; // return if they're the same
}

int main(){
    printf("%d", isSubsequence("abc", "ajdbnec"));
}